### ISSUE TITLE
allow_system_table_mods in PGOPTIONS isn't forwarded to segments

### ISSUE THREAD

**Issue by jchampio** [greenplum-db/gpdb#6703]
### Greenplum version or build

master

### Expected behavior

Setting `allow_system_table_mods` via `PGOPTIONS` should result in that GUC being forwarded to segments, as if `SET allow_system_table_mods` had been executed in SQL.

### Actual behavior

It is not set correctly on the segments, leading to some confusing behavior (see below).

### Step to reproduce the behavior

The following case assumes the existence of a TOAST table with the name `pg_toast.pg_toast_17760`.

```
$ PGOPTIONS='-c allow_system_table_mods=True' psql postgres
psql (9.4.20)
Type "help" for help.

postgres=# SHOW allow_system_table_mods;
 allow_system_table_mods
-------------------------
 on
(1 row)

postgres=# ALTER TABLE pg_toast.pg_toast_17760 RENAME TO pg_toast_renamed;
ERROR:  permission denied: "pg_toast_17760" is a system catalog  (seg0 127.0.0.1:25432 pid=61543)
postgres=# SET allow_system_table_mods=on; -- shouldn't make any difference
SET
postgres=# ALTER TABLE pg_toast.pg_toast_17760 RENAME TO pg_toast_renamed;
ALTER TABLE
```

Is this possibly related to #6610 et al?

**Comment by hlinnaka** [comment]
Yeah, we really need to re-design the way GUCs are sent to QE processes..

**Comment by zhangh43** [comment]
This issue is due to GUC 'allow_system_table_mods' didn't set flag 'GUC_GPDB_ADDOPT' and MakeOptions will not dispatch GUC without 'GUC_GPDB_ADDOPT' flag. The following patch will fix it.
```
                {"allow_system_table_mods", PGC_USERSET, CUSTOM_OPTIONS,
                        gettext_noop("Allows modifications of the structure of system tables."),
                        NULL,
-                       GUC_NOT_IN_SAMPLE | GUC_NO_SHOW_ALL
+                       GUC_NOT_IN_SAMPLE | GUC_NO_SHOW_ALL|GUC_GPDB_ADDOPT
                },
                &allowSystemTableMods,
                false,
```

But agree with Heikki, we should refactor sync GUC between qd and qe. I'll work on it.

**Comment by kainwen** [comment]
@zhangh43 
What is the state of this issue now? With the new GUC dispatch policy.

**Comment by zhangh43** [comment]
Yes It's fixed in #8533 Close this Issue.
